src/test/Test.hs

```diff
@@ -224,6 +224,7 @@ selectName typ (Names ints bools bytes) = do
                 AInteger -> ints
                 ABoolean -> bools
                 AByteStr -> bytes
+                AContract -> error "unsupported type"
```

Comment:
Is there any reason why we can't extend the generators to produce expressions involving contract types too?

Reply:
`genType` is only used to generate types of arguments, and we don't allow contract types where. This case will never actually match in the code since we never call `genType` with contract. I will add a more clear error message though.